chore(docs): add volume policy docs with example configurations (#136)
openebs/cstor-operators: docs/tutorial/volumes/policy.md

Following are list of policies that can be configured based on the requirements.

- [Replica Affinity to create a volume replica on specific pool](#replica-affinity)
- [Volume Target Pod Affinity](#volume-target-pod-affinity)
- [Volume Tunable](#volume-tunable)
- [Memory and CPU Resources QOS](#resource-request-and-limits) 
- [Toleration for target pod to ensure scheduling of pool pods on tainted nodes](#target-pod-toleration)
- [Priority class for volume target deployment](#priority-class)

Below StorageClass example contains `cstorVolumePolicy` parameter having `csi-volume-policy` name set to configured the custom policy.

```yaml
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  name: cstor-sparse-auto
provisioner: cstor.csi.openebs.io
allowVolumeExpansion: true
parameters:
  replicaCount: "1"
  cstorPoolCluster: "cspc-disk"
  cas-type: "cstor"
  fsType: "xfs"                 // default type is ext4
  cstorVolumePolicy: "csi-volume-policy"

```

If the volume policy is not created before volume provisioning and later we want to change any of the available policy it can be 

> change by editing the CStorVolumeConfig(CVC) resource as per volume bases which will be reconciled by the volume policy controller

Author: Correct, but the intension here is not confused the user with different controller name and keep it simple

> change by editing the CStorVolumeConfig(CVC) resource as per volume bases which will be reconciled by the volume policy controller

Author: anyways, change it to `cvc controller` as suggested

    kind: CStorVolume
    name: pvc-25e79ecb-8357-49d4-83c2-2e63ebd66278
    namespace: openebs
    resourceVersion: "1260"
    uid: ea6e09f2-1e65-41ab-820a-ed1ecd14873c
  policy:
    provision:
      replicaAffinity: true
    replica:
      zvolWorkers: "1"
    replicaPoolInfo:
    - poolName: cspc-sparse-lh7n
    target:
      affinity:
        requiredDuringSchedulingIgnoredDuringExecution:
        - labelSelector:
            matchExpressions:
            - key: openebs.io/target-affinity
              operator: In
              values:
              - percona
          namespaces:
          - default
          topologyKey: kubernetes.io/hostname
      auxResources:
        limits:
          cpu: 500m
          memory: 128Mi
        requests:
          cpu: 250m
          memory: 64Mi
      luWorkers: 8
      priorityClassName: system-cluster-critical
      queueDepth: "16"
      resources:
        limits:
          cpu: 500m
          memory: 128Mi
        requests:
        .
        .
        .
```
### Replica Affinity:

For StatefulSet applications, to distribute single replica volume on specific cstor pool we can use the `replicaAffinity` enabled scheduling.

> This feature should be used with delay volume binding i.e. `volumeBindingMode: WaitForFirstConsumer` in StorageClass as shown below.

Author: Yes , required for best effort.

User will need to add the following label to both Application and volume Policy.

Configured Policy having target-affinity label for example, using `kubernetes.io/hostname` as a topologyKey in CStorVolumePolicy:

```yaml
apiVersion: cstor.openebs.io/v1
kind: CStorVolumePolicy
metadata:
  name: csi-volume-policy
  namespace: openebs
spec:
  target:
    affinity:
      requiredDuringSchedulingIgnoredDuringExecution:
      - labelSelector:
          matchExpressions:
          - key: openebs.io/target-affinity
            operator: In
            values:
            - fio-cstor                              // application-unique-label
        topologyKey: kubernetes.io/hostname
        namespaces: ["default"]                      // application namespace
```

Set the label configured in volume policy created above `openebs.io/target-affinity: fio-cstor` on the app pod which will be used to find pods, by label, within the domain defined by topologyKey.

```yaml
apiVersion: v1
kind: Pod
metadata:
  name: fio-cstor
  namespace: default
  labels:
    name: fio-cstor
    openebs.io/target-affinity: fio-cstor
```

### Volume Tunable:

> Allow users to set available performance tunings in Volume Policy based on their workload. Below are the tunings that can be configured

Reviewer: How about `Policy allows users to set available performance tunings based on their workload`?

Author: done , rephrased the sentence.

- `queueDepth`:

> cStor target `queueDepth`, This limits the ongoing IO count from client. Default is 32.

Reviewer: ```suggestion
cStor target `queueDepth`, This limits the ongoing IO count from iscsi client on node to cStor target pod. The default value is 32.
```

Author: done

- `luworkers`

> cStor target worker IO threads, sets the number of threads that are working on above queue.

Reviewer: How about the below lines?
```suggestion
cStor target IO worker threads, set the number of threads that are working on the `QueueDepth` queue.
```

Default value is `6`. In case of better number of cores and RAM, this value can be `16`. 
This means 16 threads will be running for each volume.

- `zvolWorkers`:
cStor volume replica IO worker threads, defaults to the number of cores on the machine.
In case of better number of cores and RAM, this value can be `16`.

```yaml
apiVersion: cstor.openebs.io/v1
kind: CStorVolumePolicy
metadata:
  name: csi-volume-policy
  namespace: openebs
spec:
  replica:
    zvolWorkers: "4"
  target:
    luWorkers: 6
    queueDepth: "32"
```

> Note: These configuration can be changed once the volume get provisioned by editing the CStorVolumeConfig resource on per volume bases.

Reviewer: ```suggestion
Note: These configurations can be changed after provisioning volumes also by editing the corresponding volume CStorVolumeConfig resources.
```

Author: rephrased it
